generator/src/Definition/YamlReader.php

```diff
@@ -20,9 +17,8 @@ public function read(string $dirname): array
 
         $definitions = [];
         foreach ($finder as $file) {
-            $operator = Yaml::parseFile($file->getPathname());
-            assert(is_array($operator));
-            $definitions[] = new OperatorDefinition(...$operator);
+            $operator = Yaml::parseFile($file->getPathname(), Yaml::PARSE_OBJECT | Yaml::PARSE_OBJECT_FOR_MAP);
```

Comment:
The option `PARSE_OBJECT_FOR_MAP` is required to ensure the `$count` empty object `{}` from the Yaml is not converted to an empty array. `[]`